### ISSUE TITLE
look for the train and test dataset that the  hopenet used

### ISSUE THREAD

**Issue by winsonxxxx** [Cjiangbpcs/hopenet_mxJiang#1]
hi, I am student  from  China. I want to retrain the hopenet from: https://github.com/natanielruiz/deep-head-pose;  but I spend a faw time and can not download the train and test dataset than the paper used（300W-LP、AFLW2000、BIWI and so on).  Can you  send me the datasets ? my email : winsonxxxx@163.com. Thanks a lot!

**Comment by Cjiangbpcs** [comment]
Does the following link work for you?
 [300W-LP]: https://drive.google.com/file/d/0B7OEHD3T4eCkVGs0TkhUWFN6N1k/view?usp=sharing?

**Comment by winsonxxxx** [comment]
@Cjiangbpcs ,Thanks  a  lot!

**Comment by winsonxxxx** [comment]
@Cjiangbpcs , sorry to bother you angin. My retrain gluon version of hopenet does not converge like the original Pytorch version, too. The yaw、pitch、roll loss is around 7.5、4.5、4.5 in the first epoch and can not descend any more. And my train dataset is 300W_LP , batch size , lr, alpha value are as the same with the original Pytorch version code. Do you fit the converge problem and recurrence the result of the original version? Hope to discuss with you

**Comment by Cjiangbpcs** [comment]
What version of gluon script you are using? Did you use pretrained resnet50 from gluon model zoo? 

I changed the batch normalization layers to use global stats instead of batch states, that seems to make my model converge much faster. You can try that to see if it works for your script.

Also did you run your version of the test_hopenet.py? I trained my model using just one image, after 20 epochs, I use the trained weight to run test_hopenet.py, the prediction should be very close to the labels, this helped me debug a problem in my code.

**Comment by winsonxxxx** [comment]
Thanks for you quick response!

I did not use pretrained resnet50 model but I modify the resnet50 output just like the original Pytorch version: add yaw, pitch ,roll dense layer. And I wonder that , why do you divide into three network（yaw,pitch,roll）in your code ?

And I run your train code , 5 epoch the yaw,pitch,roll loss is about 7, 3, 3. Is that your code in github can recurrence the result of the original version or more better?
![image](https://user-images.githubusercontent.com/12495785/49775785-4efb0300-fd34-11e8-80ff-702a080d154c.png)

I did not use the tes_hopenet.py yet. I wirte a test_on_webcam.py to test the model. Thank you for the suggestion about the batch normalization and test_hopenet.py.

**Comment by winsonxxxx** [comment]
@Cjiangbpcs , I upload the code on https://github.com/winsonxxxx/head-pose-estimation/tree/master

**Comment by Cjiangbpcs** [comment]
I will upload a latest version of my codes shortly. 

The three models are unnecessary, I used these to debug my scripts. My new version does converge fast in the beginning but oscillate a bit later, depends on the parameters I use. Did you also use batch_size=16, lr=0.00001? I used epoch 35+ to test it on my own video, looked similar to the pytorch version, but a bit worse than the video using weights they shared such as hopenet_robust.pkl.

I also took a look at your code. Since you did not load any pretrained weights, it is not surprising for your model to converge much slower then the Pytorch one. Yours is more similar to https://github.com/haofanwang.

**Comment by winsonxxxx** [comment]
Can you upload your model,too?

batch_size=16, lr=0.00001 should be run  a long time. I test batch_size=16,32,64, lr=0.001, 0.0001, the alpha value I use 0.001, do you test alpha value ? My problem is that when I run the train script, the yaw,pitch,roll loss do not descend in the first epoch, and test the model on webcam , whether I change the pose, yaw,pitch,roll value are in a small range (-2, -5). Maybe it has bug in my code.

I also have a look at https://github.com/haofanwang,they used their own gray images for datasets, and use a smiple cnn network.

**Comment by Cjiangbpcs** [comment]
I am testing my new version and will upload it when it is ready, hopefully soon.

I use alpha=1 or 2 as in the paper. The cross entropy error is very stable, it is the mse error that fluctuates a lot. If you use alpha=0.001, you are mainly looking at the cross entropy error. I recalled the paper tests a bunch of alpha and found alpha=1 or 2 gave the best results, correct?

**Comment by winsonxxxx** [comment]
yes, the paper use alpha=1 or 2.
![image](https://user-images.githubusercontent.com/12495785/49850838-0453b680-fe1a-11e8-8601-8b4e708b3ea8.png)

Do you know this three parameters in the paper?
![image](https://user-images.githubusercontent.com/12495785/49850937-67454d80-fe1a-11e8-9b74-281d4396e037.png)

![image](https://user-images.githubusercontent.com/12495785/49850949-70ceb580-fe1a-11e8-9276-1ff1c22ad134.png)

**Comment by winsonxxxx** [comment]
I know that, the three parameters are Adam's parameters.
![image](https://user-images.githubusercontent.com/12495785/49851255-7c6eac00-fe1b-11e8-902f-c0381eaeb741.png)

**Comment by Cjiangbpcs** [comment]
yeah, those three parameters are usually set to default as above.

**Comment by jake221** [comment]
@winsonxxxx @Cjiangbpcs 
Can you reproduce the result of the Hopenet project or paper? When I train use your gluon version code, I find if I set the alpha as 0.001 I can obtain low loss and it is very stable at the very begining. But the test accuracy is very low. However, if I set the alpha as 0.1, the loss is very very high and cannot decrease. How could I solve the problem? Could you push your latest code or the default setting parameter for training? Thanks!
